CommandDotNet/ArgumentInfoCreator.cs

```diff
@@ -16,15 +17,37 @@ public ArgumentInfoCreator(AppSettings settings)
         {
             _settings = settings;
         }
+
+        public IEnumerable<ArgumentInfo> GetArgumentsFromMethod(MethodBase methodBase, ArgumentMode argumentMode)
```

Comment:
moving the logic to these methods makes it possible to reuse in test setup so we're not copying implementation logic.